src/devices/cpu/arcompact/arcompact.h

```diff
@@ -105,795 +40,751 @@ class arcompact_device : public cpu_device
+
+	uint32_t handleop32_B_cc_D_s21(uint32_t op);
+	uint32_t handleop32_B_D_s25(uint32_t op);
+	uint32_t handleop32_BL_cc_d_s21(uint32_t op);
+	uint32_t handleop32_BL_d_s25(uint32_t op);
+
+	int32_t get_01_01_01_address_offset(uint32_t op);
+	uint32_t BRxx_takejump(uint32_t address, uint8_t n, int size);
+	bool BRxx_condition(uint8_t condition, uint32_t b, uint32_t c);
+	uint32_t handleop32_BRxx_reg_reg(uint32_t op, uint8_t condition);
+	uint32_t handleop32_BRxx_reg_imm(uint32_t op, uint8_t condition);
+
+	uint32_t handleop32_BREQ_reg_reg(uint32_t op);
+	uint32_t handleop32_BRNE_reg_reg(uint32_t op);
+	uint32_t handleop32_BRLT_reg_reg(uint32_t op);
+	uint32_t handleop32_BRGE_reg_reg(uint32_t op);
+	uint32_t handleop32_BRLO_reg_reg(uint32_t op);
+	uint32_t handleop32_BRHS_reg_reg(uint32_t op);
+	uint32_t handleop32_BBIT0_reg_reg(uint32_t op);
+	uint32_t handleop32_BBIT1_reg_reg(uint32_t op);
+	uint32_t handleop32_BREQ_reg_imm(uint32_t op);
+	uint32_t handleop32_BRNE_reg_imm(uint32_t op);
+	uint32_t handleop32_BRLT_reg_imm(uint32_t op);
+	uint32_t handleop32_BRGE_reg_imm(uint32_t op);
+	uint32_t handleop32_BRLO_reg_imm(uint32_t op);
+	uint32_t handleop32_BRHS_reg_imm(uint32_t op);
+	uint32_t handleop32_BBIT0_reg_imm(uint32_t op);
+	uint32_t handleop32_BBIT1_reg_imm(uint32_t op);
+	uint32_t handleop32_LD_r_o(uint32_t op);
+	uint32_t handleop32_ST_r_o(uint32_t op);
+
+	typedef uint32_t (*ophandler32)(void*,uint32_t,uint32_t,uint8_t);
+	typedef void (*ophandler32_ff)(void*,uint32_t,uint32_t);
+	typedef void (*ophandler32_mul)(void*,uint32_t,uint32_t);
+	typedef uint32_t (*ophandler32_sop)(void*,uint32_t, uint8_t);
+
+	uint32_t handleop32_general(uint32_t op, ophandler32 ophandler);
+	uint32_t handleop32_general_MULx64(uint32_t op, ophandler32_mul ophandler);
+	uint32_t handleop32_general_nowriteback_forced_flag(uint32_t op, ophandler32_ff ophandler);
+	uint32_t handleop32_general_SOP_group(uint32_t op, ophandler32_sop ophandler);
+
+	// arcompact_execute.cpp
+	// used by both arcompact_execute_ops_04_3x.cpp and arcompact_execute_ops_02to03.cpp
+	void arcompact_handle_ld_helper(uint32_t op, uint8_t areg, uint8_t breg, uint32_t s, uint8_t X, uint8_t Z, uint8_t a);
+
+	// arcompact_execute_ops_04.cpp
+	static uint32_t handleop32_ADD_do_op(void* obj, uint32_t src1, uint32_t src2, uint8_t set_flags);
+	static uint32_t handleop32_ADC_do_op(void* obj, uint32_t src1, uint32_t src2, uint8_t set_flags);
+	static uint32_t handleop32_SUB_do_op(void* obj, uint32_t src1, uint32_t src2, uint8_t set_flags);
+	static uint32_t handleop32_AND_do_op(void* obj, uint32_t src1, uint32_t src2, uint8_t set_flags);
+	static uint32_t handleop32_OR_do_op(void* obj, uint32_t src1, uint32_t src2, uint8_t set_flags);
+
+	static uint32_t handleop32_BIC_do_op(void* obj, uint32_t src1, uint32_t src2, uint8_t set_flags);
+	static uint32_t handleop32_XOR_do_op(void* obj, uint32_t src1, uint32_t src2, uint8_t set_flags);
```

Comment:
Again, why `void *` and `uint8_t` for what should be `arcompact_device &` and `bool`?

Reply:
done